task/src/global_registry.rs

```diff
@@ -32,12 +33,12 @@ pub struct GlobalRegistry {
     /// up-to-date shared list of statuses
     /// only used if `state_cpy` is out of date
     /// or if appending
-    state_list: Arc<RwLock<Vec<(TaskState, String)>>>,
+    state_list: Arc<RwLock<BTreeMap<HotShotTaskId, (TaskState, String)>>>,
     /// possibly stale read version of state
     /// NOTE: must include entire state in order to
     /// support both incrementing and reading.
     /// Writing to the status should gracefully shut down the task
-    state_cache: Vec<(TaskState, String)>,
+    state_cache: BTreeMap<HotShotTaskId, (TaskState, String)>,
```

Comment:
`state_cache` exists to avoid contention on `state_list`. We only use `state_list` if `state_cache` lacks the entry. That way `register` or normal state operations don't block on the same lock. This should be super cheap memory wise since each tuple is a few words in memory. If we have `n` tasks, contending for the same lock will slow things down.